File: 6 kyu/Bit Counting/bitcounting.js
var countBits = function(n) {
    let bin
    ones = 0
    
    bin = n.toString(2);
    console.log(bin)
    for(let n of bin) {
        if(n == 1) {
            ones++;
        }
    }

    return ones;
  };


console.log(countBits(6170057006));

File: 6 kyu/Bit Counting/bitcounting.c
#include <stddef.h>

size_t countBits(unsigned value)
{
    int ones = 0, i = 0;
    int num;

    while (value) {
        num = value % 2;
        if(num == 1) ones++;
        i++;
        value /= 2;
    }

	return ones;
}

int main() {
    int resul = 0;

    resul = countBits(1234);
    printf("resul = %d\n", resul);

}